include/zephyr/drivers/sensor.h

```diff
@@ -899,20 +879,69 @@ static inline int z_impl_sensor_sample_fetch_chan(const struct device *dev,
  *
  * @return 0 if successful, negative errno code if failure.
  */
-__syscall int sensor_channel_get(const struct device *dev,
-				 enum sensor_channel chan,
-				 struct sensor_value *val);
+__deprecated __syscall int sensor_channel_get(const struct device *dev, enum sensor_channel chan,
+					      struct sensor_value *val);
 
-static inline int z_impl_sensor_channel_get(const struct device *dev,
-					    enum sensor_channel chan,
+static inline int z_impl_sensor_channel_get(const struct device *dev, enum sensor_channel chan,
 					    struct sensor_value *val)
 {
-	const struct sensor_driver_api *api =
-		(const struct sensor_driver_api *)dev->api;
+	const struct sensor_driver_api *api = (const struct sensor_driver_api *)dev->api;
 
 	return api->channel_get(dev, chan, val);
 }
 
+/**
+ * @param[in] dev Pointer to the sensor device
+ * @param[in] channels Channels to read
+ * @param[in] num_channels The number of channels in the channel array
+ * @param[out] shift The shift value for the result(s)
+ * @param[out] values The read values. Note that some channels produce more than one value, it's
+ *             the caller's responsibility to allocate enough memory.
+ * @param[in] num_values The number of allocated values.
+ * @return 0 if successful, negative errno code on failure.
+ */
+int sensor_read_and_decode(const struct device *dev, struct sensor_chan_spec *channels,
+			   size_t num_channels, int8_t *shift, q31_t *values, size_t num_values);
```

Comment:
I agree this is what we discussed in the last Sensors WG meeting. I do wonder, though, if this would be intuitive enough for people to handle 3-axis data: perhaps we can showcase this API used in a sample both for a TPH sensor and an Accelerometer?

Reply:
I was thinking I should go ahead and convert some of the in tree uses so we can see how this is used. I'll do that today.